Simulator/Simulator.WebJob/SimulatorModule.cs

```diff
@@ -8,66 +8,31 @@
-
-            builder.RegisterType<DeviceRegistryRepository>()
-                .As<IDeviceRegistryListRepository>();
-
-            builder.RegisterType<DeviceRulesRepository>()
-                .As<IDeviceRulesRepository>();
-
-            builder.RegisterType<SecurityKeyGenerator>()
-                .As<ISecurityKeyGenerator>();
-
-            builder.RegisterType<VirtualDeviceTableStorage>()
-                .As<IVirtualDeviceStorage>();
-
-            builder.RegisterType<ActionMappingLogic>()
-                .As<IActionMappingLogic>();
-
-            builder.RegisterType<ActionMappingRepository>()
-                .As<IActionMappingRepository>();
-
-            builder.RegisterType<ActionLogic>()
-                .As<IActionLogic>();
-
-            builder.RegisterType<DataInitializer>()
-                .As<IDataInitializer>();
-
-            builder.RegisterType<ActionRepository>()
-                .As<IActionRepository>();
-
-            builder.RegisterType<DocDbRestUtility>()
-                .As<IDocDbRestUtility>();
-
-            builder.RegisterType<AzureTableStorageClientFactory>()
-                .As<IAzureTableStorageClientFactory>();
-
-            builder.RegisterType<BlobStorageClientFactory>()
-                .As<IBlobStorageClientFactory>();
+            builder.RegisterType<ConfigurationProvider>().As<IConfigurationProvider>().SingleInstance();
+            builder.RegisterType<DeviceLogic>().As<IDeviceLogic>();
+            builder.RegisterType<IotHubRepository>().As<IIotHubRepository>();
+            builder.RegisterType<IoTHubDeviceManager>().As<IIoTHubDeviceManager>();
+            builder.RegisterType<DeviceRulesLogic>().As<IDeviceRulesLogic>();
+            builder.RegisterType<DeviceRegistryRepository>().As<IDeviceRegistryCrudRepository>();
+            builder.RegisterType<DeviceRegistryRepository>().As<IDeviceRegistryListRepository>();
+            builder.RegisterType<DeviceRulesRepository>().As<IDeviceRulesRepository>();
+            builder.RegisterType<SecurityKeyGenerator>().As<ISecurityKeyGenerator>();
+            builder.RegisterType<VirtualDeviceTableStorage>().As<IVirtualDeviceStorage>();
+            builder.RegisterType<ActionMappingLogic>().As<IActionMappingLogic>();
+            builder.RegisterType<ActionMappingRepository>().As<IActionMappingRepository>();
+            builder.RegisterType<ActionLogic>().As<IActionLogic>();
+            builder.RegisterType<DataInitializer>().As<IDataInitializer>();
+            builder.RegisterType<ActionRepository>().As<IActionRepository>();
+            builder.RegisterType<DocDbRestUtility>().As<IDocDbRestUtility>();
+            builder.RegisterType<AzureTableStorageClientFactory>().As<IAzureTableStorageClientFactory>();
+            builder.RegisterType<BlobStorageClientFactory>().As<IBlobStorageClientFactory>();
+            builder.RegisterGeneric(typeof(DocumentDBClient<>)).As(typeof(IDocumentDBClient<>));
```

Comment:
Yes. This is open generic registration. So at runtime if a class requires say IDocumentDBClient<DeviceModel> the container will new up a DocumentDBClient<> open generic as DocumentDBClient<DeviceModel>